### PR TITLE
Simplify subscriber download API

### PR DESCRIPTION
**Pull request by ivilata**
This reduces subscriber API endpoints to get dataset data to just `/api/fetch`, which will always first get missing chunks (for the given slice) or dependent datasets (for lazy expressions) and serve the result as a Blosc2 frame. With this `/api/download` and `/api/download-url` become obsolete, fetching always takes place over `/api/fetch` and that is always the download endpoint.

Redundant or obsolete subscriber API endpoints have been removed. However the API client module still makes a distinction between `fetch` and `download`: the former always returns a `numpy.ndarray` while the latter always downloads to disk (in Blosc2 frame format unless the remote file is a plain file of non-Blosc2 type).

### CODE REVIEW

File: caterva2/api_utils.py [hunk 1]
```diff
@@ -83,10 +84,8 @@ def fetch_data(path, sub_url, params, auth_cookie=None):
     return data
 
 
-def get_download_url(path, sub_url, auth_cookie=None):
-    response = _xget(f'{sub_url}api/download-url/{path}',
-                     auth_cookie=auth_cookie)
-    return response.json()
+def get_download_url(path, sub_url):
```
Comment (FrancescAlted) [inline caterva2/api_utils.py:87]: I think it is better to stay with the naming convention, so `sub_url` -> `urlbase`.

File: caterva2/clients/cli.py [hunk 2]
```diff
@@ -100,10 +100,7 @@ def cmd_list(args, auth_cookie):
 @handle_errors
 @with_auth_cookie
 def cmd_url(args, auth_cookie):
-    # TODO: provide a url that can be used to open the dataset in blosc2
-    # TODO: add a new function to the API that returns the url
-    data = api_utils.get_download_url(args.root, args.sub_url,
-                                      auth_cookie=auth_cookie)
+    data = api_utils.get_download_url(args.dataset, args.sub_url)
```
Comment (FrancescAlted) [inline caterva2/clients/cli.py:103]: Ditto

File: caterva2/services/sub.py [hunk 3]
```diff
@@ -504,14 +473,21 @@ async def fetch_data(
...
+        The (slice of) dataset as a Blosc2 schunk.
     """
 
     slice_ = api_utils.parse_slice(slice_)
     # Download and update the necessary chunks of the schunk in cache
     abspath, dataprep = abspath_and_dataprep(path, slice_, user=user)
     await dataprep()
 
+    # TODO: Also slices covering all data (step 1 or none).
+    whole = slice_ is None or slice_ == () or slice_ == slice(None)
```
Comment (FrancescAlted) [inline caterva2/services/sub.py]: For the TODO, maybe something along this (not tested)? 
```
if isinstance(slice_, tuple) and len(slice_) > 1: 
    whole *= all(s.step in (None, 1) for s in slice_ if isinstance(s, slice)) 
```

Comment (ivilata) [reply caterva2/services/sub.py]: Implemented the TODO in commit c8fbecb.

File: caterva2/api_utils.py (hunk 1 above)
Comment (ivilata) [reply caterva2/api_utils.py:87]: I can see that, but  doing that rename in this PR would clutter it and be beyond its purpose.

**Review by FrancescAlted — APPROVED** [review]
LGTM

**Pull request merged**